### DISCUSSION TITLE
Windows filesystem not mounted

### DISCUSSION THREAD

**Discussion by sirwobin** [nix-community/NixOS-WSL#121] [Q&A] [answered]
/mnt/c exists but is empty.  /etc/fstab is likewise empty hence `mount /mnt/c` fails with the error that it can't be found in fstab.  Being new to Nix I'm not sure how further to diagnose this problem.  I have an ubuntu distro which mounts /mnt/c fine and access works well.  How can I achieve the same level of access to Windows drives from within Nix?

Contents of /etc/wsl.conf as follows:
```
[automount]
enabled=1
mountFsTab=1
options=metadata,uid=1000,gid=100
root=/mnt/
```

I am in awe of how good it is to use Nix under WSL.  Amazing work!  Getting host FS access inside nix would be the cherry on top.

How I built my distro:
- I installed NixOS-WSL from https://github.com/nix-community/NixOS-WSL/releases/download/21.11-d89f18a1/nixos-wsl-installer-fixed.tar.gz
- In Windows 10 Powershell executed `wsl --import NixOS .\NixOS\ nixos-wsl-installer-fixed.tar.gz --version 2`
- I added 2 lines to /etc/nixos/configuration.nix to enable zsh and make zsh default for nixos user.
- I added the unstable channel then executed `nixos-rebuilt switch --upgrade`
- nixos-version reports "22.11pre398667.39d7f929fbc (Raccoon)"

**Comment by nzbr** [comment]
In theory, it should work just like with any other WSL distro.
I just tested upgrading to nixos-unstable (`22.11pre399062.3a11db5f408 (Raccoon)`) and the files in `/mnt/c` are still present. It's completely normal for `/etc/fstab` to be empty, because the windows drives are mounted by WSL's `/init` and not by the distro itself

Can you test if it works on a freshly imported copy?

**Reply by sirwobin** [reply]
Fresh copy unpacked and /mnt/c is empty.  No other actions performed before checking.

**Comment by sirwobin** [comment, marked as answer]
Without me doing anything else to remedy this it appears to have fixed itself after some months.

**Comment by volkswagenfeature** [comment]
Hey, this popped up for me as well, but I think I'm running the 47ab671 release, so might not be relevant at this time. Will reply if I discover it on the latest release. To mount a windows filesystem manually, the command is `sudo mount -t drvfs C: /mnt/c`
